src/ert/storage/local_storage.py

```diff
@@ -137,8 +155,17 @@ def __exit__(
 class LocalStorageAccessor(LocalStorageReader):
     LOCK_TIMEOUT = 5
 
-    def __init__(self, path: Union[str, os.PathLike[str]]) -> None:
-        super().__init__(path)
+    def __init__(
+        self,
+        path: Union[str, os.PathLike[str]],
+        *,
+        ignore_migration_check: bool = False,
+    ) -> None:
+        self.path = Path(path)
+        if not ignore_migration_check and local_storage_needs_migration(self.path):
+            from ert.storage.migration.block_fs import migrate  # pylint: disable=C0415
+
+            migrate(self.path)
```

Comment:
Not used at the moment, perhaps turn this migration into an issue instead? I would suggest moving this to the main entry point instead, so we have more control of the migration.

Reply:
It would be used in production (if storage needs migration, migrate)

Reply:
Should be optional though? So you get a prompt, saying would you like to delete or migrate?